src/transaction.js

```diff
@@ -52,23 +52,12 @@ Transaction.fromBuffer = function (buffer, __noStrict) {
 
   var vinLen = readVarInt()
   for (var i = 0; i < vinLen; ++i) {
-    var hash = readSlice(32)
-
-    if (Transaction.isCoinbaseHash(hash)) {
```

Comment:
Since `script` is now a buffer,  we don't need to treat it differently for generation scripts.